Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -1531,40 +1545,48 @@ void TreeResolver::sortPositionOptionsIfNeeded(PositionOptions& options, const S
         return;
     options.sorted = true;
 
-    auto order = options.originalStyle->positionTryOrder();
-    if (order == PositionTryOrder::Normal || options.optionStyles.size() < 2)
-        return;
```

Comment:
Seems weird to enter sorting function with nothing to sort. Can't the part the needs to run always be in the caller?

Reply:
Yeah, that's why I was referring to. Can't the thing that needs to be done always be done in the caller?

Reply:
The adjustment for last successful fallback (rotate the position options array such that the last successful fallback is the first element) can only be done after sorting the options, and it should only be done once. If we do the adjustment in `tryChoosePositionOption` then we'd need a flag (or use the `sorted` flag) to ensure we only adjust once. So I thought might as well put it in `sortPositionOptionsIfNeeded`.

Technically we could avoid the array rotation by starting the index at the first option, and increment it (wrap around when at the end) until we got to the first option again. But that'd require remembering which option is the first one. To make it simple, I just make it the convention that the first option is at index 0, and rotate the array as needed.